File: ios/MyNativeMethods.swift
//
//  MyNativeMethods.swift
//  NativeModuleExample
//
//  Created by Noel Escobedo on 12/13/17.
//  Copyright © 2017 Facebook. All rights reserved.
//

import Foundation

@objc(MyNativeMethods)
class MyNativeMethods: NSObject {
  
  @objc(calcAverage:callback:)
  func calcAverage(data_array: NSArray, callback: RCTResponseSenderBlock) -> Void {
    // Date is ready to use!
    NSLog("%@ %@ %@", data_array) // Esto solo hace un print de los datos en la consola de Xcode
    //let ret: [String : Any] = ["data_array": data_array]
    let lenght: Int = data_array.count;
    var total: Int = 0;
    for value in data_array {
      total += value as! Int;
    }
    let averange: Int = total / lenght;
    
    callback([averange, NSNull()]) //La primera posición es una respuesta correcta y la segunda es un error
  }
  
  open static func requiresMainQueueSetup() -> Bool {
    return true;
  }
  
}

File: android/app/src/main/java/com/nativemoduleexample/mymodules/MyNativeMethods.java
package com.nativemoduleexample.mymodules;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReadableArray;

/**
 * Created by noel on 12/13/17.
 */

public class MyNativeMethods extends ReactContextBaseJavaModule {
    public MyNativeMethods(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "MyNativeMethods";
    }

    @ReactMethod
    public void calcAverage(ReadableArray data_array, Callback responseCallback) {
        int length = data_array.size();
        int total = 0;
        for (int i = 0; i < length; i++) {
            total += data_array.getInt(i);
        }
        int averange = total / length;
        responseCallback.invoke(averange);
    }
}
